src/server/actions/indicator.ts

```
@@ -0,0 +1,104 @@
+
+function getClosingValues(olhcvList: number[][], fromTimestamp?: number): number[] {
+  // [timestamp, open, high, low, close , volume] // this is the format of the olhcvList
+  return fromTimestamp
+    ? olhcvList.filter(([timestamp]) => timestamp >= fromTimestamp).map(([, , , , close]) => close)
+    : olhcvList.map(([, , , , close]) => close);
+}
+
+function calculateIndicator(indicator: string, closingValues: number[], parameters: IndicatorParameters): number {
+  switch (indicator) {
+    case 'rsi':
+      const rsiResult = rsi(closingValues, { period: parameters.period || DEFAULT_PERIOD });
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)